worker/create_batches_from_filters_worker.go

```diff
@@ -23,265 +23,137 @@
+func (b *CreateBatchesFromFiltersWorker) createDbToCsvJob(job *model.Job, page DBPage,
+	uploader *s3.CreateMultipartUploadOutput, total int) {
 	filters := job.Filters
 	whereClause := GetWhereClauseFromFilters(filters)
-	limit := b.DBPageSize
 	var query string
 	if (whereClause) != "" {
-		query = fmt.Sprintf("SELECT user_id FROM %s WHERE seq_id > %d AND %s ORDER BY seq_id ASC LIMIT %d;", GetPushDBTableName(job.App.Name, job.Service), page.Offset, whereClause, limit)
+		query = fmt.Sprintf("SELECT DISTINCT user_id FROM %s WHERE user_id > '%s' AND user_id <= '%s' AND %s ORDER BY user_id;", GetPushDBTableName(job.App.Name, job.Service), page.SmallestID, page.BiggestID, whereClause)
 	} else {
-		query = fmt.Sprintf("SELECT user_id FROM %s WHERE seq_id > %d ORDER BY seq_id ASC LIMIT %d;", GetPushDBTableName(job.App.Name, job.Service), page.Offset, limit)
+		query = fmt.Sprintf("SELECT DISTINCT user_id FROM %s WHERE user_id > '%s' AND user_id <= '%s' ORDER BY user_id;", GetPushDBTableName(job.App.Name, job.Service), page.SmallestID, page.BiggestID)
 	}
-	var users []User
-	start := time.Now()
-	_, err := b.PushDB.DB.Query(&users, query)
-	labels := []string{fmt.Sprintf("game:%s", job.App.Name), fmt.Sprintf("platform:%s", job.Service)}
-	b.Workers.Statsd.Timing("get_page_with_filters", time.Now().Sub(start), labels, 1)
-
-	checkErr(b.Logger, err)
-	return &users
+	b.Workers.DbToCsvJob(&ToCSVMenssage{
+		Query:      query,
+		PartNumber: page.Page,
+		Uploader:   *uploader,
+		TotalJobs:  total,
+		Job:        *job,
+	})
 }
 
-func (b *CreateBatchesFromFiltersWorker) preprocessPages(job *model.Job, stageStatus *StageStatus) ([]DBPage, int, int) {
+func (b *CreateBatchesFromFiltersWorker) preprocessPages(job *model.Job) ([]DBPage, int, int) {
 	filters := job.Filters
 	var count int
 	whereClause := GetWhereClauseFromFilters(filters)
 	var query string
 	if (whereClause) != "" {
-		query = fmt.Sprintf("SELECT count(1) FROM %s WHERE %s;", GetPushDBTableName(job.App.Name, job.Service), whereClause)
+		query = fmt.Sprintf("SELECT count(DISTINCT user_id) FROM %s WHERE %s", GetPushDBTableName(job.App.Name, job.Service), whereClause)
 	} else {
-		query = fmt.Sprintf("SELECT count(1) FROM %s;", GetPushDBTableName(job.App.Name, job.Service))
+		query = fmt.Sprintf("SELECT count(DISTINCT user_id) FROM %s", GetPushDBTableName(job.App.Name, job.Service))
 	}
-	_, err := b.PushDB.DB.Query(&count, query)
+
+	labels := []string{fmt.Sprintf("game:%s", job.App.Name), fmt.Sprintf("platform:%s", job.Service)}
+	start := time.Now()
+
+	_, err := b.Workers.PushDB.DB.Query(&count, query)
 	if count == 0 {
 		checkErr(b.Logger, fmt.Errorf("no users matching the filters"))
 	}
-	pageCount := int(math.Ceil(float64(count) / float64(b.DBPageSize)))
+	b.Workers.Statsd.Timing("count_total", time.Now().Sub(start), labels, 1)
+
+	// The maximum of ids to fill one multiupload part (in mb)
```

Comment:
we don't have any guarantees that user_id is an uuid, in fact there are several non-uuid ids in our environment

Reply:
Already changed to use a string has the id.